cardano-launcher/src/Cardano/Shell/Launcher.hs

```diff
@@ -125,6 +153,22 @@ instance Isomorphism DaedalusExitCode ExitCode where
 -- Functions
 --------------------------------------------------------------------------------
 
+readSafeMode :: FilePath -> IO WalletMode
+readSafeMode stateDir = do
+  let safeModeConfigFile = getSafeModeConfigPath stateDir
+  decoded <- liftIO $ Y.decodeFileEither safeModeConfigFile
+  case decoded of
+    Right value -> pure $ smcSafeMode value
+    Left _      -> pure WalletModeNormal
+
+saveSafeMode :: FilePath -> WalletMode -> IO ()
+saveSafeMode stateDir mode = do
+  let safeModeConfigFile = getSafeModeConfigPath stateDir
+  Y.encodeFile safeModeConfigFile $ SafeModeConfig mode
+
+getSafeModeConfigPath :: FilePath -> FilePath
+getSafeModeConfigPath stateDir = stateDir </> "safemode.yaml"
```

Comment:
I have to admit that this looks pretty weird. The state is being persisted into a `*.yaml` config?

Reply:
i used yaml instead of a flat file, so we have the option to add more fields in the future